source/interface/transport_interface.h

```diff
@@ -96,16 +96,31 @@
  *                                      size_t bytesToRecv )
  * {
  *     int32_t bytesReceived = 0;
- *     bytesReceived = TLSRecv( pNetworkContext->tlsContext,
- *                              pBuffer,
- *                              bytesToRecv,
- *                              MY_SOCKET_TIMEOUT );
- *     if( bytesReceived < 0 )
+ *     bool callTlsRecvFunc = true;
+ *
+ *     // For a single byte read request, check if data is available on the network.
+ *     if( bytesToRecv == 1 )
  *     {
- *         // Handle socket error.
+ *        // If no data is available on the network, do not call TLSRecv
+ *        // to avoid blocking for socket timeout.
+ *        if( TLSRecvCount( pNetworkContext->tlsContext ) == 0 )
+ *        {
+ *            callTlsRecvFunct = false;
```

Comment:
```suggestion
 *            callTlsRecvFunc = false;
```

Reply:
Also why can you just shorten this into a an OR condition like https://github.com/FreeRTOS/FreeRTOS/pull/452/files#diff-988c68b1458ff5f8870190348da08357275ca5b5286dfd6776a236166829b113R131

Reply:
For easier readability for the reader